src/test/java/com/shanks/weaving/WeavingApplicationTests.java

```diff
@@ -24,9 +29,16 @@ void contextLoads() {
 
 	@Test
 	void countQueries(){
+		entityManagerFactory.getCache().evictAll();
 		QueryCountHolder.clear();
-		this.projectRepository.findAllById(List.of(1,2));
+		this.postRepository.findAllById(List.of(1,2));
 		var count = QueryCountHolder.getGrandTotal().getSelect();
-		Assertions.assertEquals(3, count);
+		QueryCountHolder.clear();
+		Assertions.assertEquals(1, count);
+	}
+
+	// comment below function and above test will pass.
```

Comment:
another weird behavior is that presence of below unused private method somehow interferes with weaving. i.e. if method below is commented, weaving would happen and test should pass.

Reply:
Also some additional references around spring and load time weaving using aspectj/spring-instrument. ( [ref](https://docs.spring.io/spring-framework/reference/core/aop/using-aspectj.html#aop-aj-ltw) )